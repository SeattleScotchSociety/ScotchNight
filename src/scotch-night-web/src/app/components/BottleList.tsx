import { inject, observer } from "mobx-react";
import * as React from "react";
import * as _ from 'lodash';

import { IRootStore } from "../stores/RootStore";

interface IBottleListProps { store: IRootStore; }

@inject("store")
@observer
export class BottleList extends React.Component<IBottleListProps> {
    constructor(props: IBottleListProps) {
        super(props);

        this._handleOnClick = this._handleOnClick.bind(this);
    }

    private _handleOnClick(bottle) {
        const { navigation, scotchNightStore } = this.props.store;

        scotchNightStore.setCurrentBottle(bottle);
        navigation.push(`/bottles/${bottle.id}`);
    }

    public render() {
        const { bottles } = this.props.store.bottleStore;
        const bottleElements = _.sortBy(bottles, ['distillery', 'name']).map((bottle) => {
            const { name, distillery } = bottle;

            return (
                <div className="card" key={bottle.id.toString()} onClick={() => this._handleOnClick(bottle)}>
                    <h3 className="card__head">{distillery}</h3>
                    <h4 className="card__subhead">{name}</h4>
                </div>
            );
        });

        return (
            <div className="container">
                <h1 className="margin-bottom-md">Scotch Library</h1>
                {bottleElements}
            </div>
        );
    }
}

export default BottleList;
