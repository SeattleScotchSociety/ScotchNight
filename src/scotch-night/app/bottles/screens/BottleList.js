// @flow
import React from 'react';
import { StyleSheet, Button, View, FlatList, ActivityIndicator } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loadBottles } from '../BottleActions';
import BottleListItem from '../components/BottleListItem';

class BottleList extends React.Component {
    static navigationOptions = {
        title: 'Bottles'
    };

    componentWillMount() {
        let { loadBottles } = this.props.actions;

        loadBottles();
    }

    renderHeader = () => {
        return (
            <View style={{ flex: 1 }}>
                <Button title="Add Bottle" onPress={this._onAddPress.bind(this)} />
            </View>
        );
    };

    renderFooter = () => {
        if (!this.props.loading) return null;

        return (
            <View
                style={{
                    paddingVertical: 20,
                    borderTopWidth: 1,
                    borderColor: '#CED0CE'
                }}>
                <ActivityIndicator animating size="large" />
            </View>
        );
    };

    _onAddPress() {
        const { navigate } = this.props.navigation;

        navigate('AddBottle', {});
    }

    render() {
        const { navigate } = this.props.navigation;

        return (
            <View style={{ flex: 1 }}>
                <FlatList
                    data={this.props.bottles || []}
                    renderItem={({ item }) => (
                        <BottleListItem
                            bottle={{ ...item }}
                            onPress={() => navigate('BottleDetail', { bottle: { ...item } })}
                        />
                    )}
                    keyExtractor={item => item.id}
                    ListHeaderComponent={this.renderHeader}
                    ListFooterComponent={this.renderFooter}
                />
            </View>
        );
    }
}

function mapStateToProps(state) {
    let { bottles } = state;

    return { bottles };
}

function mapDispatchToProps(dispatch) {
    return { actions: bindActionCreators({ loadBottles }, dispatch) };
}

export default connect(mapStateToProps, mapDispatchToProps)(BottleList);

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center'
    }
});
