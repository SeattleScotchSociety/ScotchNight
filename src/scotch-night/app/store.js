import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { AsyncStorage } from 'react-native';
import { persistStore, autoRehydrate } from 'redux-persist';
import { reducer } from './Reducers';
import rootSaga from './Sagas';
import Bottles from './bottles';
let { loadBottles } = Bottles.Actions;

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
    reducer,
    undefined,
    compose(applyMiddleware(sagaMiddleware), autoRehydrate())
);

sagaMiddleware.run(rootSaga);

persistStore(store, { storage: AsyncStorage }, () => {
    store.dispatch(loadBottles());
}).purge();

export default store;
