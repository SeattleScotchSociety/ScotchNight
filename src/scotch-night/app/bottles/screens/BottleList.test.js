import React from 'react';
import renderer from 'react-test-renderer';
import Store from '../../store';
import BottleList from './BottleList';

test('renders correctly', () => {
    const tree = renderer.create(<BottleList store={Store} navigation={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
});
