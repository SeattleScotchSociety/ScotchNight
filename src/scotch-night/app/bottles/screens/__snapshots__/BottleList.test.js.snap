// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ScrollView
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "bottleImageUrl": "http://www.westlanddistillery.com/assets/media/whiskeys/5e4a5c3c6cdbfabdcf67941c83f9e7d9.jpg",
        "bottleUrl": "http://www.westlanddistillery.com/whiskey/peated-malt",
        "description": "I bought this because I like smoke.",
        "distillery": "Westland",
        "distilleryUrl": "http://www.westlanddistillery.com",
        "id": 1,
        "member": "Adam Phillabaum",
        "memberRatings": Array [
          Object {
            "member": "Shannon Andrews",
            "notes": "Reminds me of my trip to Islay",
            "rating": "4",
          },
          Object {
            "member": "Travis Merkel",
            "notes": "I own 27 bottles of this!",
            "rating": "5",
          },
        ],
        "name": "Peated",
      },
      Object {
        "distillery": "Laphroaig",
        "id": 2,
        "name": "18 Year",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onViewableItemsChanged={undefined}
  removeClippedSubviews={true}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  shouldItemUpdate={[Function]}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Add Bottle
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "#ededed",
              "borderBottomWidth": 1,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 16.099999999999998,
                    },
                    Object {
                      "marginLeft": 10,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Westland - Peated
            </Text>
          </View>
          <View
            style={undefined}
          >
            <View />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0.15,
              "justifyContent": "center",
            }
          }
        >
          <View
            onPress={undefined}
            style={
              Array [
                false,
                false,
                false,
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            underlayColor="white"
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#bdc6cf",
                    "fontSize": 28,
                  },
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              î—Œ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "#ededed",
              "borderBottomWidth": 1,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#43484d",
                      "fontSize": 16.099999999999998,
                    },
                    Object {
                      "marginLeft": 10,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Laphroaig - 18 Year
            </Text>
          </View>
          <View
            style={undefined}
          >
            <View />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0.15,
              "justifyContent": "center",
            }
          }
        >
          <View
            onPress={undefined}
            style={
              Array [
                false,
                false,
                false,
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            underlayColor="white"
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#bdc6cf",
                    "fontSize": 28,
                  },
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              î—Œ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 0,
      }
    }
  />
  <View
    onLayout={[Function]}
  />
</ScrollView>
`;
